src/Renderer.cpp

```diff
@@ -1309,6 +1313,36 @@ auto RendererRender(
             g_globalUniforms.CameraDirection = glm::vec4(cameraDirection, aspectRatio);
             UpdateBuffer(g_globalUniformsBuffer, 0, sizeof(TGpuGlobalUniforms), &g_globalUniforms);
         });
+*/
+
+        registry.view<TComponentCamera, TComponentTransform, TComponentOrientationEuler>().each([&](
+            const auto& entity,
+            const auto& cameraComponent,
+            auto& transformComponent,
+            const auto& orientationEulerComponent) {
+
+            auto globalTransform = EntityGetGlobalTransform(registry, entity);
+            glm::vec3 globalPosition;
+            glm::vec3 globalScale;
+            glm::quat globalOrientation;
+            glm::vec4 globalPerspective;
+            glm::vec3 globalSkew;
+
+            glm::decompose(globalTransform, globalScale, globalOrientation, globalPosition, globalSkew, globalPerspective);
+
+            glm::quat orientation = glm::eulerAngleYX(orientationEulerComponent.Yaw, orientationEulerComponent.Pitch);
+            transformComponent = glm::translate(glm::mat4(1.0f), globalPosition) * glm::mat4_cast(orientation);
+
+            auto aspectRatio = g_scaledFramebufferSize.x / static_cast<float>(g_scaledFramebufferSize.y);
+            auto cameraDirection = glm::normalize(orientation * glm::vec3(0.0, 0.0, -1.0));
+            g_globalUniforms.ProjectionMatrix = glm::infinitePerspective(glm::radians(cameraComponent.FieldOfView), aspectRatio, 0.1f);
+            g_globalUniforms.ViewMatrix = glm::inverse(transformComponent);
+
+            g_globalUniforms.CameraPosition = glm::vec4(globalPosition, glm::radians(cameraComponent.FieldOfView));
+            g_globalUniforms.CameraDirection = glm::vec4(cameraDirection, aspectRatio);
```

Comment:
Is decomposing necessary? Seems like a lot of code for what could just be

```cpp
glm::mat4 cameraMatrix = EntityGetGlobalTransform(registry, entity);
glm::mat4 viewMatrix = glm::inverse(cameraMatrix);
glm::vec3 cameraPos = cameraMatrix[3];
glm::vec3 cameraDirection = cameraMatrix[1];
```

Reply:
Decomposing is absolutely not necessary. I didn't see that I can just do that.